src/components/LoginDialog/LoginDialog.vue

```diff
@@ -0,0 +1,104 @@
+<template>
+  <div class="text-center">
+    <v-dialog v-model="dialog" width="500" v-if="!currentUser">
+      <template v-slot:activator="{ on, attrs }">
+        <v-btn fab small outlined v-bind="attrs" v-on="on">
+          <v-icon> mdi-account </v-icon>
+        </v-btn>
+      </template>
+
+      <v-card>
+        <v-card-title class="text-h5"> Login </v-card-title>
+
+        <v-card-text>
+          <v-form ref="form" v-model="valid">
+            <v-text-field
+              prepend-icon="mdi-account"
+              name="email"
+              label="Email"
+              type="text"
+              required
+              v-model="email"
+              :rules="emailRules"
+            ></v-text-field>
+            <v-text-field
+              id="password"
+              prepend-icon="mdi-lock"
+              name="password"
+              label="Password"
+              type="password"
+              v-model="password"
+              :rules="passwordRules"
+            ></v-text-field>
+          </v-form>
+        </v-card-text>
+
+        <v-card-actions>
+          <v-spacer></v-spacer>
+          <v-btn color="primary" @click="login" :disabled="!valid">Login</v-btn>
+        </v-card-actions>
+      </v-card>
+    </v-dialog>
+
+    <v-menu offset-y v-else>
+      <template v-slot:activator="{ on, attrs }">
+        <v-btn fab small outlined v-bind="attrs" v-on="on">
+          <v-icon> mdi-account-tie </v-icon>
+        </v-btn>
+      </template>
+      <v-list class="login-dialog__user-actions-list">
+        <v-list-item @click="logout">
+          <v-list-item-title>Log Out</v-list-item-title>
+        </v-list-item>
+      </v-list>
+    </v-menu>
+  </div>
+</template>
+
+<script>
+import { mapState, mapActions } from "vuex";
+
+export default {
+  data() {
+    return {
+      dialog: false,
+      valid: false,
+      email: "",
+      emailRules: [
+        (v) => !!v || "E-mail is required",
+        (v) => /.+@.+\..+/.test(v) || "E-mail must be valid",
+      ],
+      password: "",
+      passwordRules: [
+        (v) => !!v || "Password is required",
+        (v) => (v && v.length >= 8) || "Password must be at least 8 characters",
+      ],
+    };
+  },
+  computed: mapState(["currentUser"]),
+  methods: {
+    ...mapActions(["setCurrentUser", "removeCurrentUser"]),
```

Comment:
I usually use mapMutations instead of mapActions to set data directly into the store. Is it a best practice to have an action wrapper for mutations? 🤔

Reply:
I don't know what are the best practices, it is just what we did before and also what is used on the RWS-Viewer. In any case it's probably not worth changing it everywhere now, but you can ask Sjoerd and let me know.